src/tools/perf/perftest.c

```diff
@@ -343,16 +344,22 @@ static ucs_status_t setup_sock_rte_p2p(struct perftest_context *ctx)
         goto out;
     }
 
+    if (res == NULL) {
+        snprintf(err_str, 64, "getaddrinfo() returned empty list");
+    }
+
     for (t = res; t != NULL; t = t->ai_next) {
         sockfd = socket(t->ai_family, t->ai_socktype, t->ai_protocol);
         if (sockfd < 0) {
+            snprintf(err_str, 64, "socket() failed: %s", strerror(errno));
```

Comment:
minor: can use %m instead of %s to get the error string, then no need to use strerror